server/v2/store/server.go

```diff
@@ -63,15 +64,29 @@ func (s *Server[T]) CLICommands() serverv2.CLIConfig {
 			s.ListSnapshotsCmd(),
 			s.DumpArchiveCmd(),
 			s.LoadArchiveCmd(),
-			s.RestoreSnapshotCmd(s.appCreator),
+			s.RestoreSnapshotCmd(s.backend),
 		},
 	}
 }
 
 func (s *Server[T]) Config() any {
 	if s.config == nil || s.config.AppDBBackend == "" {
-		return DefaultConfig()
+		return root.DefaultConfig()
 	}
 
 	return s.config
 }
+
+func UnmarshalConfig(cfg map[string]any) *root.Config {
+	config := &root.Config{
+		Options: root.DefaultStoreOptions(),
+	}
+	if err := serverv2.UnmarshalSubConfig(cfg, ServerName, config); err != nil {
+		panic(fmt.Sprintf("failed to unmarshal config: %v", err))
+	}
+	home := cfg["home"]
```

Comment:
If home is nil, I think we should return an error instead right?

Reply:
I think so too. I'll adjust the tests which have this as nil to something else.

Reply:
hmm I have store/v2 erroring if empty, so we'll produce an error if it's ever used, which seems appropriate. 
https://github.com/cosmos/cosmos-sdk/blob/ad5b9caa3cd8eea5c8aff276e6486d02c16d805c/store/v2/root/builder.go#L44
i'm inclined to keep as is for now since this just handles parsing?

Reply:
okay, then adding a comment that says that would be useful 👍🏾